src/App.js

```diff
@@ -1,75 +1,189 @@
+          title: prevState.title,
+          completed: false,
+        }],
+        title: '',
+        selectedFilter: FILTERS[0],
+      };
+    });
+  };
+
+  deleteTodo = (todoId) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.filter(todo => todo.id !== todoId),
+    }));
+  };
+
+  clearAllCompleted = () => {
+    this.setState(prevState => ({
+      todos: prevState.todos.filter(todo => !todo.completed),
+    }));
+  };
+
+  toggleTodoCompleted = (todoId) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((todo) => {
+        if (todo.id !== todoId) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }),
+    }));
+  };
+
+  toggleAllComplete = () => {
+    this.setState((prevState) => {
+      if (prevState.todos.every(todo => todo.completed)) {
+        return {
+          todos: prevState.todos.map(todo => ({
+            ...todo,
+            completed: false,
+          })),
+        };
+      }
+
+      return {
+        todos: prevState.todos.map(todo => ({
+          ...todo,
+          completed: true,
+        })),
+      };
+    });
+  };
 
-      <footer className="footer" style={{ display: 'block' }}>
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button
-          type="button"
-          className="clear-completed"
-          style={{ display: 'block' }}
-        />
-      </footer>
-    </section>
-  );
+  setFilter = (filter) => {
+    this.setState({
+      selectedFilter: filter,
+    });
+  };
+
+  filterTodos = () => {
+    switch (this.state.selectedFilter) {
+      case FILTERS[1]:
+        return this.state.todos.filter(todo => !todo.completed);
+      case FILTERS[2]:
+        return this.state.todos.filter(todo => todo.completed);
+      default:
+        return this.state.todos.filter(todo => todo.id);
+    }
+  }
+
+  render = () => {
+    const { todos, title, selectedFilter } = this.state;
+    const completedTodos = todos.filter(todo => !todo.completed).length;
+    const visibleTodos = this.filterTodos();
+    const isAllCompleted = todos.every(todo => todo.completed);
+
+    return (
+      <section className="todoapp">
+        <header className="header">
+          <h1>todos</h1>
+
+          <form onSubmit={this.addTodo}>
+            <input
+              className="new-todo"
+              placeholder="What needs to be done?"
+              type="text"
+              onChange={this.setTitle}
+              value={title}
+            />
+          </form>
+
+        </header>
+
+        {todos.length === 0 || (
```

Comment:
зазвичай пишуть ось так 
```suggestion
        {todos.length > 0 && (
```
up to you